File: Java/Plus One.java
// 0ms, 100.00%; 37.9MB, 5.64%
class Solution {
    public int[] plusOne(int[] digits) {
        int carry = 1;
        for (int i = digits.length - 1; i > -1; --i) {
            digits[i] += carry;
            if (digits[i] > 9) {
                digits[i] -= 10;
                carry = 1;
            } else {
                carry = 0;
            }
        }
        if (carry == 0) {
            return digits;
        }
        int[] _new = new int[digits.length + 1];
        _new[0] = carry;
        System.arraycopy(digits, 0, _new, 1, digits.length);
        return _new;
    }
}

File: Go/Plus One.go
// 0ms, 100.00%; 2.1MB, 100.00%
func plusOne(digits []int) []int {
	for i := len(digits) - 1; i > -1; i-- {
		if digits[i] < 9 {
			digits[i]++
			return digits
		}
		digits[i] = 0
	}
	arr := make([]int, len(digits)+1)
	arr[0] = 1
	return arr
}
